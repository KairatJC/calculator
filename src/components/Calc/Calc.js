import React from "react";
import "./Calc.css";
import CalcContext from "../../CalcContext";

function Calc() {
  return (
    <CalcContext.Consumer>
      {value => {
        let result = getResult(value);
        return (
          <div className="calc wrap">
            {result}
          </div>
        );
      }}
    </CalcContext.Consumer>
  );
}

function getResult(value) {
  if (value.operator !== null || value.oldValue !== null) {
    return (
      <div className="calc-display">
        <span>
          {value.oldValue}
        </span>
        <span>
          {value.operator}
        </span>
        <span>
          {value.currentValue}
        </span>
      </div>
    );
  } else {
    return (
      <div className="calc-display">
        <span>
          {value.currentValue}
        </span>
      </div>
    );
  }
}

export default Calc;
