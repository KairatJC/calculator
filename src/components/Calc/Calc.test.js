import React from "react";
import renderer from "react-test-renderer";
import Calc from "./Calc";

test("Link changes the class when hovered", () => {
  const component = renderer.create(<Calc />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
