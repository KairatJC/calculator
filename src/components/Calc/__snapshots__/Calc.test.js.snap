// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="calc wrap"
>
  <div
    className="calc-display"
  >
    <span />
  </div>
</div>
`;
